# The workforce planning

1. Data processing

In [2]:
import pandas as pd
import plotly.express as px

In [3]:
data = pd.read_csv('/content/Workforce_Skills_Gap_Analysis.csv')

**Cost-Benefit Analysis:** Determine the financial impact of skill shortage and compare it with the average training cost per employee for a cost-effective reskilling strategy.

In [11]:
df = data.copy()

# Calculate the total training cost by multiplying 'Employees Needing Reskilling' with 'Average Training Cost Per Employee'
df['Total Training Cost (USD Million)'] = (df['Employees Needing Reskilling (Thousands)'] * df['Average Training Cost Per Employee (USD)']) / 1000

# Calculate the net financial impact of skill shortage by subtracting 'Total Training Cost' from 'Business Impact of Skill Shortage'
df['Net Financial Impact (USD Million)'] = df['Business Impact of Skill Shortage (USD Million)'] - df['Total Training Cost (USD Million)']

# Group by 'Year' and calculate the average of 'Net Financial Impact' for each year
df = df.groupby('Year')['Net Financial Impact (USD Million)'].mean().reset_index()

# Sort the DataFrame by 'Net Financial Impact (USD Million)'
df = df.sort_values(by='Year', ascending=False)
df

Year  Net Financial Impact (USD Million)
4  2023                         -677.707556
3  2022                         -876.271500
2  2021                         -760.597500
1  2020                         -801.146667
0  2019                         -806.601750

2. Data visualization

In [12]:
fig = px.bar(df,
             x='Year',
             y='Net Financial Impact (USD Million)',
             title="Net Financial Impact Over Years",
             labels={'Net Financial Impact (USD Million)':'Net Financial Impact (Million USD)',
                     'Year':'Year'},
             color='Net Financial Impact (USD Million)',
             color_continuous_scale='Rainbow')

fig.update_layout(template='plotly_dark')


fig.show()

The data analysis reveals a consistent negative financial impact due to skill shortage over the years, with an average net impact of approximately -784.5 USD Million per year. This consistent negative trajectory suggests a significant cost associated with skill shortage.


In 2019, the net financial impact was -806.60175 USD Million.

In 2020, it slightly decreased to -801.146667 USD Million.

A further decrease was observed in 2021 with -760.5975 USD Million.

However, the highest financial impact was seen in 2022 with -876.2715 USD Million.

In 2023, the impact reduced significantly to -677.707556 USD Million.

Comparing these figures with the average training cost per employee could provide insights for a more cost-effective reskilling strategy.

In [4]:
df = data.copy()
df['Year'] = pd.to_datetime(df['Year'], format='%Y')

# Filter data to include only years up to current date
df = df[df['Year'] <= '2024-06-08']

# Aggregate data to get total employees needing reskilling and total business impact by country and industry
df = df.groupby(['Country', 'Industry']).agg({'Employees Needing Reskilling (Thousands)': 'sum',
                                              'Business Impact of Skill Shortage (USD Million)': 'sum'}).reset_index()

df

Country        Industry  Employees Needing Reskilling (Thousands)  \
0   Australia       Education                                       346   
1   Australia         Finance                                       452   
2   Australia   Manufacturing                                        78   
3   Australia          Retail                                       218   
4      Brazil          Energy                                       284   
5      Brazil          Retail                                       554   
6      Brazil  Transportation                                       444   
7      Canada    Construction                                       232   
8      Canada       Education                                       116   
9      Canada          Energy                                       331   
10     Canada         Finance                                       438   
11     Canada      Healthcare                                       177   
12     Canada          Retail                                       752   
13     Canada      Technology                                       643   
14     Canada  Transportation                                       361   
15      China    Construction                                       139   
16      China       Education                                       475   
17      China          Energy                                       359   
18      China          Retail                                       141   
19     France          Energy                                        59   
20     France     Hospitality                                       396   
21     France          Retail                                       398   
22     France      Technology                                       378   
23    Germany      Healthcare                                       847   
24      India    Construction                                       578   
25      India       Education                                       100   
26      Japan          Energy                                       424   
27      Japan         Finance                                       202   
28      Japan      Healthcare                                       239   
29      Japan     Hospitality                                       530   
30      Japan          Retail                                       573   
31      Japan      Technology                                       117   
32         UK       Education                                       183   
33         UK         Finance                                       787   
34         UK   Manufacturing                                       466   
35         UK          Retail                                       483   
36        USA    Construction                                       486   
37        USA   Manufacturing                                       335   
38        USA          Retail                                       142   
39        USA      Technology                                       176   

    Business Impact of Skill Shortage (USD Million)  
0                                             144.0  
1                                              46.5  
2                                              16.5  
3                                              91.5  
4                                               7.5  
5                                             204.0  
6                                             147.0  
7                                              73.5  
8                                               9.0  
9                                             150.0  
10                                             84.0  
11                                            130.5  
12                                             75.0  
13                                             81.0  
14                                            150.0  
15                                            112.5  
16          

2. Data visualization

In [5]:
fig = px.bar(df, x='Employees Needing Reskilling (Thousands)', y='Country',
             color='Industry', title='Workforce Reskilling Needs by Country and Industry',
             labels={'Employees Needing Reskilling (Thousands)': 'Employees Needing Reskilling',
                     'Country': 'Country', 'Industry': 'Industry'},
             hover_data=['Business Impact of Skill Shortage (USD Million)'])
fig.update_layout(template='plotly_dark')
fig.show()

3. Insights generation

The data reveals a significant need for reskilling across various industries and countries.

* In Australia, the finance sector has the highest reskilling need (452k employees), though the education sector experiences the most significant business impact ($144M).

* Brazil's retail and transportation sectors are most affected, with reskilling needs of 554k and 444k employees, respectively,

 and business impacts of $204M  and  $147M.

* In Canada, the retail sector presents the most reskilling needs (752k employees), while the energy and transportation sectors have the highest business impacts ($150M each).

* The healthcare sector in Germany has the highest reskilling needs (847k employees) and business impact ($261M) globally,

 followed by Japan's hospitality sector (530k employees, $279M business impact).